packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -0,0 +1,185 @@
+     * @param _initialFundingRate Starting funding rate multiplier.
+     * @param _fpFinderAddress Finder used to discover financial-product-related contracts.
+     * @param _timerAddress Contract that stores the current time in a testing environment.
+     * @param _identifier Unique identifier for DVM price feed ticker for child financial contract.
+     * Must be set to 0x0 for production environments that use live time.
+     */
+    constructor(
+        FixedPoint.Unsigned memory _initialFundingRate,
+        address _fpFinderAddress,
+        address _timerAddress,
+        bytes32 _identifier
+    ) public Testable(_timerAddress) nonReentrant() {
+        // TODO: Should we constrain `_initialFundingRate > x && < y`?
+        cumulativeFundingRateMultiplier = _initialFundingRate;
+        fpFinder = FinderInterface(_fpFinderAddress);
+        lastUpdateTime = getCurrentTime();
+        identifier = _identifier;
+    }
+
+    /****************************************
+     *         PUBLIC FUNCTIONS           *
+     ****************************************/
+
+    /**
+     * @notice Fetches the latest funding rate from the Store, scales it over the time period since the last update,
+     * and uses this effective rate to set a new funding rate multiplier.
+     * @dev A funding rate of 1.0 reported by the Store implies a neutral funding rate, meaning that the current multiplier
+     * should not change.
+     */
+    function applyFundingRate() public {
+        _applyEffectiveFundingRatePerToken();
+    }
+
+    /****************************************
+     *         INTERNAL FUNCTIONS           *
+     ****************************************/
+
+    // Returns a token amount scaled by the current funding rate multiplier.
+    // Note: if the contract has paid fees since it was deployed, the raw
+    // value should be larger than the returned value.
+    function _getFundingRateAppliedTokenDebt(FixedPoint.Unsigned memory rawTokenDebt)
+        internal
+        view
+        returns (FixedPoint.Unsigned memory tokenDebt)
+    {
+        return rawTokenDebt.mul(cumulativeFundingRateMultiplier);
+    }
+
+    function _getFundingRateStore() internal view returns (FundingRateStoreInterface) {
+        return FundingRateStoreInterface(fpFinder.getImplementationAddress("FundingRateStore"));
+    }
+
+    // Fetches a funding rate from the Store, determines the period over which to compute an effective fee,
+    // and multiplies the current multiplier by the effective fee.
+    // A funding rate < 1 will reduce the multiplier, and a funding rate of > 1 will increase the multiplier.
+    // Note: 1 is set as the neutral rate because there are no negative numbers in FixedPoint, so we decide to treat
+    // values < 1 as "negative".
+    function _applyEffectiveFundingRatePerToken() internal {
+        uint256 currentTime = getCurrentTime();
+        uint256 paymentPeriod = currentTime.sub(lastUpdateTime);
```

Comment:
Hm this is mostly to be consistent with how timing is done in the rest of the contract. What's nice is that Solidity provides a `now` keyword that returns the current time. I'd recommend we stick with this because it would require a larger refactor

Reply:
Yeah, this is something the crypto community goes back and forth on, and I prefer time for anything that relates to interest. People think about interest and time in terms of seconds rather than blocks. If we use blocks to compute interest or funding payments, then the APY in terms of real time changes based on the average block time, which is not ideal IMO.